code/game/atoms_movable.dm

```diff
@@ -164,23 +151,62 @@
 	INVOKE_ASYNC(src, .proc/SpinAnimation, 5, 2)
 	return TRUE
 
+///Move a movable between z levels, to a target turf (skips can_z_move checks), or in an ascending or descending direction IF valid.
+/atom/movable/proc/zMove(dir, turf/target, feedback = FALSE, forced = FALSE, affect_pulling = TRUE)
+	if(!target)
+		if(dir == UP || dir == DOWN)
+			target = get_step_multiz(src, dir)
+		if(!target)
+			if(feedback)
+				to_chat(src, "<span class='warning'>There's nowhere to go in that direction!</span>")
+			return FALSE
+		if(!can_z_move(dir, get_turf(src), target, forced ? NONE : ZTRAVEL_CAN_FLY_CHECKS))
+			if(feedback)
+				to_chat(src, "<span class='warning'>You couldn't move there!</span>")
+			return FALSE
+	var/list/fallen_heroes = list(src)
+	if(affect_pulling && pulling)
+		fallen_heroes += pulling
+		if(pulling.buckled_mobs)
+			fallen_heroes += pulling.buckled_mobs
```

Comment:
`Pulling a mob who's buckled to a something` vs `pulling something with a mob buckled to it` behaves slightly differently, do both situations work with this?

Reply:
Umh, nice consideration. I'll give it a test and ascertain they both work equally fine.

Reply:
do they?

Reply:
Yes, but on the local working tree, not the latest pushed commit here. I shall update this PR this weekend when I have more free time.